docs/configure/integration.md

```diff
@@ -0,0 +1,315 @@
+
+#### Using Your Existing Config
+
+If you have an existing webpack config for your project and want to reuse this app's configuration, you can import your main webpack config into Storybook's [`.storybook/main.js`](./overview#configure-story-rendering) and merge both:
+
+The following code snippet shows how you can replace the loaders from Storybook with the ones from your app's `webpack.config.js`:
+
+```js
+// .storybook/main.js
+const path = require('path');
+
+// your app's webpack.config.js
+const custom = require('../webpack.config.js');
+
+module.exports = {
+  webpackFinal: (config) => {
+    return { ...config, module: { ...config.module, rules: custom.module.rules } };
+  },
+};
+```
+
+### Babel
+
+Storybook’s webpack config by [default](#default-configuration) sets up [Babel](https://babeljs.io/) for ES6 transpiling. Storybook works with evergreen browsers and IE11 by default.
+
+Here are some key features of Storybook's Babel configurations.
+
+#### Default configuration
+
+We have added ES2016 support with Babel for transpiling your JS code.
+
+In addition to that, we've added a few additional features, like object spreading and async await.
+
+Check out our [source](https://github.com/storybookjs/storybook/blob/master/lib/core/src/server/common/babel.js) to learn more about these plugins.
+
+#### Custom configuration
+
+If your project has a `.babelrc` file, we'll use that instead of the default config file.
+
+You can also place a `.storybook/.babelrc` file to use a special configuration for Storybook only.
+
+
+### TypeScript
+
+Storybook has built-in Typescript support, so your Typescript project should work with zero configuration needed.
+
+#### Default configuration
+
+The base Typescript configuration uses [`babel-loader`](https://webpack.js.org/loaders/babel-loader/) for Typescript transpilation, and optionally [`fork-ts-checker-webpack-plugin`](https://github.com/TypeStrong/fork-ts-checker-webpack-plugin) for checking.
+
+Each framework uses the base configuration unless otherwise specified:
+
+- Angular ignores the base and uses `ts-loader` and `ngx-template-loader`.
+- Vue ignores the uses `ts-loader` and applies it to both `.tsx` and `.vue` files.
+- React adds `react-docgen-typescript-plugin` to the base.
+
+
+#### Main.js configuration
+
+To make it easier to configure Typescript handling, use the `typescript` field in your [`.storybook/main.js`](./overview#configure-story-rendering).
+
+The following code snippets shows the fields for you to use with TypeScript:
+
+```js
+// .storybook/main.js
+module.exports = {
+  typescript: {
+    check: false,
+    checkOptions: {},
+    reactDocgen: 'react-docgen-typescript',
+    reactDocgenTypescriptOptions: {
+      shouldExtractLiteralValuesFromEnum: true,
+      propFilter: (prop) => (prop.parent ? !/node_modules/.test(prop.parent.fileName) : true),
+    },
+  },
+};
+```
+
+<div style="background-color:#F8FAFC">
+TODO: vet SEE DOCS link
```

Comment:
@shilman ?

Reply:
https://github.com/TypeStrong/fork-ts-checker-webpack-plugin (link 1)
https://github.com/hipstersmoothie/react-docgen-typescript-plugin (link 2)

Reply:
@shilman thanks for the links. Already added.